### PR TITLE
Should get terms should return term objects?

### PR DESCRIPTION
**Pull request by mattheu**

### CODE REVIEW

**Comment by joehoyle** [comment]
Yes I think so

**Comment by joehoyle** [comment]
I guess set_terms() should expect an array of `Term`s too?

**Comment by mattheu** [comment]
Yeah. 

Also instead of passing append there could be methods for `add_term` and `reset_terms`?

**Comment by mattheu** [comment]
Updated to make better use of term objects

**Comment by joehoyle** [comment]
How do I easily remove a term?

**Comment by mattheu** [comment]
Updated to addmethod for removing.
Improved reset terms.

**Comment by mattheu** [comment]
@joehoyle this is ready for merge - any further comments?

File: wordpress-objects.post.php [hunk 1]
```diff
@@ -248,24 +248,82 @@ public function add_comment( $comment_text, $user_id ) {
...
-	 * @param string $tags     Optional. The tags to set for the post, separated by commas. Default empty.
-	 * @param string $taxonomy Optional. Taxonomy name. Default 'post_tag'.
-	 * @param bool   $append   Optional. If true, don't delete existing tags, just add on. If false,
-	 *                         replace the tags with the new tags. Default false.
-	 * @return mixed Array of affected term IDs. WP_Error or false on failure.
+	 * @param array $terms Term objects
 	 */
-	public function set_terms( $tags, $taxonomy, $append = false ) {
-		return wp_set_post_terms( $this->get_id(), $tags, $taxonomy, $append );
+	public function add_terms( $terms ) {
```
Comment (joehoyle) [inline wordpress-objects.post.php]: @mattheu This should probably call `wp_set_object_terms` directly as you can pass all at once which I think will only cause one insert query.

Comment (mattheu) [inline wordpress-objects.post.php]: Updated so it is does a single insert query.

**Comment by joehoyle** [comment]
Thanks @mattheu !

**Pull request merged**